compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -164,10 +166,129 @@ static SmallVector<int64_t> getCanonicalVectorShape(func::FuncOp funcOp) {
   return canonicalVectorShape;
 }
 
+/// Tries to infer the vector sizes from an IR that has been tiled for
+/// vectorization. Returns failure if vector sizes can't be inferred. For know
+/// this utility is only able to extract tile size information from
+/// `tensor.extract_slice` and `affine.min` ops, which should cover most but not
+/// all the cases.
+/// TODO(dcaballe)" Enhance/replace this logic with `FlatAffineConstraints`
+/// utilities.
+static FailureOr<SmallVector<int64_t>> inferVectorSizesFromIR(
+    linalg::LinalgOp linalgOp) {
+  LDBG("Getting vector sizes for:\n" << linalgOp << "\n");
+
+  auto idxMaps = linalgOp.getIndexingMapsArray();
+  assert(!idxMaps.empty());
+  unsigned numDims = idxMaps[0].getNumDims();
+
+  SmallVector<int64_t> vectorSizes;
+  for (int dim = 0; dim < numDims; ++dim) {
+    // Map dimension `dim` to an operand dimension that we will use to
+    // traverse the U-D chain to get `dim` vector size information.
+    Value operand;
+    unsigned operandDim;
+    if (failed(linalgOp.mapIterationSpaceDimToOperandDim(dim, operand,
+                                                         operandDim))) {
+      return failure();
+    }
+
+    // Static case: `dim` size is available in the operand type.
+    int64_t dimSize =
+        operand.getType().cast<ShapedType>().getShape()[operandDim];
+    if (!ShapedType::isDynamic(dimSize)) {
+      vectorSizes.push_back(dimSize);
+      continue;
+    }
+
+    // Dynamic case: Traverse the U-D chain of the operand and try to find an
+    // operation (e.g., `tensor.extract_slice`, `affine.min`) that contains
+    // static information about the `dim` size.
+    Operation *currentOp = operand.getDefiningOp();
+    while (currentOp) {
+      LDBG("Inspecting for vector sizes: " << *currentOp << "\n");
+
+      if (isa<AffineMinOp>(currentOp)) {
+        break;
+      }
+      if (auto extractSliceOp = dyn_cast<tensor::ExtractSliceOp>(currentOp)) {
+        if (!extractSliceOp.isDynamicSize(operandDim)) {
+          // Extract slice has static information about the `dim` size.
+          break;
+        }
+        currentOp = extractSliceOp.getDynamicSize(operandDim).getDefiningOp();
+        continue;
+      }
+      if (auto subviewOp = dyn_cast<memref::SubViewOp>(currentOp)) {
+        if (!subviewOp.isDynamicSize(operandDim)) {
+          // Subview has static information about the `dim` size.
+          break;
+        }
+        currentOp = subviewOp.getDynamicSize(operandDim).getDefiningOp();
+        continue;
+      }
+      if (auto affineApplyOp = dyn_cast<AffineApplyOp>(currentOp)) {
+        AffineValueMap valueMap = affineApplyOp.getAffineValueMap();
+        if (valueMap.getNumDims() != 1 || valueMap.getNumSymbols() != 1)
+          return failure();
+        currentOp = valueMap.getOperand(1).getDefiningOp();
```

Comment:
Why are you restricting it to 1 dimension and 1 symbol?

Reply:
And I am not sure what `getOperand(1)` does for an `AffineValueMap`

Reply:
I'll add a comment that should answer your question